coderd/templates_test.go

```diff
@@ -1618,6 +1618,69 @@ func TestPatchTemplateMeta(t *testing.T) {
 		require.NoError(t, err)
 		assert.False(t, updated.UseClassicParameterFlow, "expected false")
 	})
+
+	t.Run("SupportEmptyOrDefault", func(t *testing.T) {
+		t.Parallel()
+
+		client := coderdtest.New(t, nil)
+		user := coderdtest.CreateFirstUser(t, client)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+
+		description := "test-description"
+		icon := "/icon/icon.png"
+		defaultTTLMillis := 10 * time.Hour.Milliseconds()
+
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID, func(ctr *codersdk.CreateTemplateRequest) {
+			ctr.Description = description
+			ctr.Icon = icon
+			ctr.DefaultTTLMillis = ptr.Ref(defaultTTLMillis)
+		})
+
+		restoreReq := codersdk.UpdateTemplateMeta{
+			Description:      &description,
+			Icon:             &icon,
+			DefaultTTLMillis: defaultTTLMillis,
+		}
+
+		type expected struct {
+			description      string
+			icon             string
+			defaultTTLMillis int64
+		}
+
+		type testCase struct {
+			name     string
+			req      codersdk.UpdateTemplateMeta
+			expected expected
+		}
+
+		tests := []testCase{
+			{name: "Only update default_ttl_ms", req: codersdk.UpdateTemplateMeta{DefaultTTLMillis: 99 * time.Hour.Milliseconds()}, expected: expected{description: template.Description, icon: template.Icon, defaultTTLMillis: 99 * time.Hour.Milliseconds()}},
+			{name: "Clear description", req: codersdk.UpdateTemplateMeta{Description: ptr.Ref("")}, expected: expected{description: "", icon: template.Icon, defaultTTLMillis: 0}},
+			{name: "Clear icon", req: codersdk.UpdateTemplateMeta{Icon: ptr.Ref("")}, expected: expected{description: template.Description, icon: "", defaultTTLMillis: 0}},
+			{name: "Nil description defaults to template description", req: codersdk.UpdateTemplateMeta{Description: nil}, expected: expected{description: template.Description, icon: template.Icon, defaultTTLMillis: 0}},
+			{name: "Nil icon defaults to template icon", req: codersdk.UpdateTemplateMeta{Icon: nil}, expected: expected{description: template.Description, icon: template.Icon, defaultTTLMillis: 0}},
+		}
+
+		// It is unfortunate we need to sleep, but the test can fail if the
+		// updatedAt is too close together.
+		time.Sleep(time.Millisecond * 5)
+
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				ctx := testutil.Context(t, testutil.WaitLong)
+				updated, err := client.UpdateTemplateMeta(ctx, template.ID, tc.req)
+				require.NoError(t, err)
+				assert.Equal(t, tc.expected.description, updated.Description)
+				assert.Equal(t, tc.expected.icon, updated.Icon)
+				assert.Equal(t, tc.expected.defaultTTLMillis, updated.DefaultTTLMillis)
+
+				// Restore template after each test case
+				_, err = client.UpdateTemplateMeta(ctx, template.ID, restoreReq)
+				require.NoError(t, err)
+			})
```

Comment:
Just make sure the `restore` always runs, even if the original context is cancelled, the original update errors, or anything else crazy happens.
```suggestion
			t.Run(tc.name, func(t *testing.T) {
				defer func() {
					ctx := testutil.Context(t, testutil.WaitShort)
					// Restore template after each test case
					_, err := client.UpdateTemplateMeta(ctx, template.ID, restoreReq)
					require.NoError(t, err)
				}()
				
				ctx := testutil.Context(t, testutil.WaitLong)
				updated, err := client.UpdateTemplateMeta(ctx, template.ID, tc.req)
				require.NoError(t, err)
				assert.Equal(t, tc.expected.description, updated.Description)
				assert.Equal(t, tc.expected.icon, updated.Icon)
				assert.Equal(t, tc.expected.defaultTTLMillis, updated.DefaultTTLMillis)
			})
```